Vsevolod-Trusov/nestmodule
# Node-develop: task 1-6 done (rename postman-file extension as  postman_collection_deploy.json)

Task 1: Base: 

- create server
- implement basic request, which sends html-response
- deploy on render

Task 2: CRUD:

- Implement get, post, put, delete methods with mocked data
- set up postman queries

Task 3: Validation

- use class-validator for validation input data

Task 4: Database MongoDB

- use atlas to interaction with mongoDB
- use mongoose package to set up it 
- create entity
- use architecture patterns as repository and unit-of-work(data-service in project)
-  write middleware that check database connection when every request to notes controller goes

Task 5: JWT

- implement access and refresh token for authentication
- implement necessary auth service and strategies
- implement guards to use it above controller's methods

Task 6: Env

- set up config module earlier when set up jwt
- add necessary env-fields
- add it in .gitignore
- add logic in 'create note' service-method to use these fields

Link to test: https://nodemodule-kpfu.onrend
Postman request-collection( !RENAME FILE AS postman_collection_deploy.json ): 

[postman_collection_deploy.zip](https://github.com/Vsevolod-Trusov/nestmodule/files/12002619/postman_collection_deploy.zip)
